Lib/gzip.py

```diff
@@ -403,6 +403,58 @@ def __iter__(self):
         return self._buffer.__iter__()
 
 
+def _read_exact(fp, n):
+    '''Read exactly *n* bytes from `fp`
+
+    This method is required because fp may be unbuffered,
+    i.e. return short reads.
+    '''
+    data = fp.read(n)
+    while len(data) < n:
+        b = fp.read(n - len(data))
+        if not b:
+            raise EOFError("Compressed file ended before the "
+                           "end-of-stream marker was reached")
+        data += b
+    return data
+
+
+def _read_gzip_header(fp):
+    '''Read a gzip header from a filestream and progresses the stream to
+    the end of the header. Returns last mtime if header was present or None
+    if no header was present'''
```

Comment:
Per PEP 257, we want a single introductory line in a docstring before longer prose.

```suggestion
    '''Read a gzip header from `fp` and progress to the end of the header.
    
    Returns last mtime if header was present or None otherwise.
    '''
```